Platform-Linux/Hardening/SSH-Config/1-Ansible/SECURE_SSH_COPY_KEY_AND_UPDATES.yaml

```diff
@@ -0,0 +1,258 @@
+# Usage
+# This script will copy your public key and enforce PubkeyAuthentication by disabling password authentication 
+# and ensuring pubkeyauthentication is enabled
+
+# Instructions
+# Make sure the lookup path to your public key is correct.
+
+# For CCDC 2025, 
+# - this script's credit is given to Alex, was copied over to this repo(CCDC2025) by Chisom
+
+
+---
+- name: Copy SSH keys and configure sshd_config
+  hosts: all
+  become: true
+
+  # Global vars, change here
+  vars:
+    current_user_name: "alex"
+    ssh_password_authentication: "no"
+    ssh_permit_empty_passwords: "no"
+    ssh_use_pam: "no"
+    ssh_permit_root_login: "no"
+    ssh_pubkey_authentication: "yes"
+    ssh_port: "22"
+    package_list_all:
+      - zsh
+      - git
+      - wget
+      - vim
+    specific_package_list_ubuntu:
+      - zsh
+    specific_package_list_centos:
+      - zsh
+    #manual_installations:
+
+  tasks:
+    - name: Copy SSH public key
+      authorized_key:
+        user: "alex"
+        key: "{{ lookup('file', '~/.ssh/id_ed25519.pub') }}"
+        state: present
```

Comment:
Maybe make this more generic @P-rple I know Alex wrote it but you copied it over : )